### PR TITLE
Add params sync pass 

### PR DESCRIPTION
**Pull request by NHZlX**
When using analysis predictor, we load parameters, programs first, and then many pass will be processed to modify our program and parameters.  **It is worth noting that these modifications are made in the CPU**. So, if we load the parameters into the gpu, these pass will not be executed. If we load the parameters into the cpu, although these pass can be used, but the model will slow down because of the parameter coping from cpu to gpu during the inference. 

So, this pr is to handle the above problem.

### CODE REVIEW

File: paddle/fluid/inference/analysis/passes/ir_params_sync_among_devices_pass.cc [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+  auto *scope = argument->scope_ptr();
+  // Get the program which has been processed by several passes.
+  analysis_program_.reset(
+      new framework::ProgramDesc(argument->ir_analyzed_program()));
+
+  const auto &global_block = analysis_program_->Block(0);
+
+  // sync the params from cpu to gpu.
+  for (auto &var : global_block.AllVars()) {
```
Comment (Superjomn) [inline paddle/fluid/inference/analysis/passes/ir_params_sync_among_devices_pass.cc]: The global_block currently doesn't contain the new parameters that user created in passes, those parameters will remain in CPU place, and copied every time to GPU.

Comment (NHZlX) [reply paddle/fluid/inference/analysis/passes/ir_params_sync_among_devices_pass.cc]: `analysis_program_ ` contains the new parameters created by the user.

File: paddle/fluid/inference/analysis/passes/ir_params_sync_among_devices_pass.cc [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+namespace paddle {
+namespace {
+bool IsPersistable(const framework::VarDesc *var) {
+  if (var->Persistable() &&
+      var->GetType() != framework::proto::VarType::FEED_MINIBATCH &&
+      var->GetType() != framework::proto::VarType::FETCH_LIST) {
+    return true;
+  }
+  return false;
+}
```
Comment (Xreki) [inline paddle/fluid/inference/analysis/passes/ir_params_sync_among_devices_pass.cc]: 这个函数不是io.cc里面有实现吗？不用重复实现吧

Comment (NHZlX) [reply paddle/fluid/inference/analysis/passes/ir_params_sync_among_devices_pass.cc]: Done

File: paddle/fluid/inference/analysis/passes/ir_params_sync_among_devices_pass.cc (hunk 1 above)
Comment (NHZlX) [reply paddle/fluid/inference/analysis/passes/ir_params_sync_among_devices_pass.cc]: Done

**Review by Superjomn — APPROVED** [review]
LGTM

**Pull request merged**